toucan_connectors/aircall/aircall_connector.py

```diff
@@ -101,16 +107,61 @@ class AircallDataSource(ToucanDataSource):
 class AircallConnector(ToucanConnector):
     """
     This is a connector for [Aircall](https://developer.aircall.io/api-references/#endpoints)
-    using [Bearer.sh](https://app.bearer.sh/)
+    using oAuth for authentication
     """
 
+    _auth_flow = 'oauth2'
+    auth_flow_id: Optional[str]
     data_source_model: AircallDataSource
-    bearer_integration = 'aircall_oauth'
-    bearer_auth_id: str
+
+    def __init__(self, **kwargs):
+        super().__init__(
+            **{k: v for k, v in kwargs.items() if k not in OAuth2Connector.init_params}
+        )
+        self.__dict__['_oauth2_connector'] = OAuth2Connector(
+            name=kwargs['name'],
+            authorization_url=AUTHORIZATION_URL,
+            scope=SCOPE,
+            token_url=TOKEN_URL,
+            **{k: v for k, v in kwargs.items() if k in OAuth2Connector.init_params},
+        )
+
+    def build_authorization_url(self):
+        return self.__dict__['_oauth2_connector'].build_authorization_url()
+
+    def retrieve_tokens(self, authorization_response: str):
```

Comment:
Should we add a docstring to explain why you had to pass explicitly client_id and _secret?